### PR TITLE
feat: Add import reader for json

### PR DESCRIPTION
**Pull request by bigsheeper**
This PR implements a new json reader for import.

issue: https://github.com/milvus-io/milvus/issues/28521

### CODE REVIEW

**Comment by xiaocai2333** [comment]
/lgtm
add more unit test

File: pkg/util/typeutil/schema.go [hunk 1]
```diff
@@ -816,6 +816,16 @@ func GetPartitionKeyFieldSchema(schema *schemapb.CollectionSchema) (*schemapb.Fi
 	return nil, errors.New("partition key field is not found")
 }
 
+// GetDynamicField returns the dynamic field if it exists.
+func GetDynamicField(schema *schemapb.CollectionSchema) (*schemapb.FieldSchema, error) {
```
Comment (czs007) [inline pkg/util/typeutil/schema.go:820]: Change the return value from `error` to `Bool?`.

File: internal/util/importutilv2/json/validator.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	}
+	// combine the redundant pairs into dynamic field(if it has)
+	err := v.combineDynamicRow(dynamicValues, row)
+	if err != nil {
+		return nil, err
+	}
+	return row, err
+}
+
+func (v *validator) combineDynamicRow(dynamicValues map[string]any, row Row) error {
```
Comment (czs007) [inline internal/util/importutilv2/json/validator.go:106]: Only cases 6 and 7 are supported; all others will result in an error.

File: internal/util/importutilv2/json/validator.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	}
+
+	return &validator{
+		name2FieldID: name2FieldID,
+		pkField:      pkField,
+		dynamicField: dynamicField,
+	}, nil
+}
+
+func (v *validator) Validate(raw any) (Row, error) {
```
Comment (czs007) [inline internal/util/importutilv2/json/validator.go:69]: Move this into Parser to avoid memory copying

**Pull request merged**